### PR TITLE
Replace 0/0 hack with match

### PR DESCRIPTION
**Pull request by Schaeff**
Allow expressions on lhs of match arms

### CODE REVIEW

File: src/constant_evaluator/mod.rs [hunk 1]
```diff
@@ -113,13 +113,7 @@ impl<'a> Evaluator<'a> {
...
             BinaryOperator::Sub => left - right,
             BinaryOperator::Mul => left * right,
-            BinaryOperator::Div => {
-                if left == 0.into() {
-                    0.into()
-                } else {
-                    left.integer_div(right)
-                }
-            }
+            BinaryOperator::Div => left.integer_div(right),
```
Comment (chriseth) [inline src/constant_evaluator/mod.rs:116]: Does this panic in release mode?

File: src/riscv/compiler.rs [hunk 2]
```diff
@@ -217,7 +217,7 @@ instr xor Y, Z -> X {
 }
 
 pil{
-	macro is_nonzero(X) { X / X }; // 0 / 0 == 0 makes this work...
+	macro is_nonzero(X) { X / X };
```
Comment (chriseth) [inline src/riscv/compiler.rs:220]: Why did you leave this?

**Comment by leonardoalt** [comment]
Requires changes.

File: src/constant_evaluator/mod.rs (hunk 1 above)
Comment (Schaeff) [reply src/constant_evaluator/mod.rs]: Yes, added a test

**Pull request merged**